apps/projects/routes.py

```diff
@@ -176,100 +377,814 @@ def get(self, project_id):
+            preview_thumbnail = doc['preview_thumbnail']
+            if preview_thumbnail:
+                app.fs.delete(preview_thumbnail['storage_id'])
+
+            app.mongo.db.projects.delete_one({'_id': format_id(project_id)})
+            return json_response(status=204)
+        else:
+            return json_response(status=500)
+
+
+class ThumbnailsTimelineProject(MethodView):
+    SCHEMA_THUMBNAILS = {
+        'amount': {
+            'type': 'integer',
+            'required': True,
+            'empty': False,
+        },
+    }
+
+    def get(self, project_id):
+        """
+        Get video thumbnails.
+        Generate new thumbnails if null or `amount` argument different from current total thumbnails
+        ---
+        consumes:
+        - application/json
+        parameters:
+        - in: path
+          name: project_id
+          type: string
+          required: True
+          description: Unique project id
+        - name: amount
+          in: query
+          type: integer
+          description: number thumbnails to create
+        responses:
+          200:
+            description: OK
+            schema:
+              type: object
+              properties:
+                processing:
+                  type: boolean
+                  example: True
+                thumbnails:
+                  type: object
+                  example: {}
+        """
+        # validate user-agent
+        check_user_agent()
+
+        amount = request.args.get('amount', 40, int)
+        if amount <= 0:
+            amount = 40
+
+        doc = app.mongo.db.projects.find_one({'_id': format_id(project_id)})
+        if not doc:
+            return not_found("Project with id: {} was not found.".format(project_id))
+
+        # Only get thumbnails when list thumbnail 've not created yet (empty) and video is not processed any task
+        data = doc.get('thumbnails')
+        if (not data or not data.get(str(amount))) \
+                and doc.get('processing') is False:
+            # Update processing is True when begin edit video
+            app.mongo.db.projects.update_one(
+                {'_id': doc['_id']},
+                {'$set': {
+                    'processing': True,
+                    'thumbnails': {}
+                }}
+            )
+            doc['processing'] = True
+            doc['thumbnails'] = {}
+
+            # Delete all old thumbnails
+            thumbnails = []
+            for key in doc['thumbnails'].keys():
+                thumbnails = doc['thumbnails'][str(key)]
+            for thumbnail in thumbnails:
+                app.fs.delete(thumbnail['storage_id'])
+            # Run get list thumbnails of timeline for video in celery
+            task_get_list_thumbnails.delay(json_util.dumps(doc), amount)
+        return json_response({"processing": doc.get('processing'), "thumbnails": doc['thumbnails']})
+
+
+class GetRawVideoThumbnail(MethodView):
+    def get(self, project_id):
```

Comment:
I have added it but removed it later because swagger ui could not play video response (though display image was fine), it displayed huge byte string instead and made browser slow. So is it fine to show GetRawVideoThumbnail docs but can't run "try it out"?

Reply:
Yes, it's better to document it anyway. Personally I don't use swagger when I need to test endpoint, I use postman or curl or HTTPie.